tests/integration/test_cli_derivatives.py

```diff
@@ -205,3 +206,98 @@ def test_custom_imaging_sessions_created_for_missing_session_labels(
 
     # Note: order of items does not matter for dict comparison
     assert custom_ses_completed_pipes == completed_pipes_for_missing_ses_sub
+
+
+def test_unrecognized_pipelines_and_versions_excluded_from_output(
+    runner,
+    test_data,
+    test_data_upload_path,
+    default_derivatives_output_path,
+    load_test_json,
+):
+    """
+    Test that when a subset of pipelines or versions from a processing status file are unrecognized,
+    they are excluded from the output JSONLD with informative warnings, without causing the derivatives command to fail.
+    """
+    with pytest.warns(UserWarning) as w:
+        result = runner.invoke(
+            bagel,
+            [
+                "derivatives",
+                "-t",
+                test_data / "proc_status_unrecognized_pipelines.tsv",
+                "-p",
+                test_data_upload_path / "example_synthetic.jsonld",
+                "-o",
+                default_derivatives_output_path,
+            ],
+            catch_exceptions=False,
+        )
+
+    assert result.exit_code == 0, f"Errored out. STDOUT: {result.output}"
+
+    assert len(w) == 2
+    warnings = [warning.message.args[0] for warning in w]
+    for warn_substrings in [
+        ("unrecognized pipelines", "unknown-pipeline"),
+        ("unrecognized versions", "{'fmriprep': ['unknown.version']}"),
+    ]:
+        assert any(
+            all(substr in warning for substr in warn_substrings)
+            for warning in warnings
+        )
+
+    output = load_test_json(default_derivatives_output_path)
+
+    sessions_with_completed_pipes = {}
+    for sub in output["hasSamples"]:
+        if sub["hasLabel"] == "sub-01":
+            for ses in sub["hasSession"]:
+                if (
+                    ses["schemaKey"] == "ImagingSession"
+                    and "hasCompletedPipeline" in ses
+                ):
+                    sessions_with_completed_pipes[ses["hasLabel"]] = ses[
+                        "hasCompletedPipeline"
+                    ]
+
+    assert sessions_with_completed_pipes.keys() == {"ses-01"}
+    ses01_completed_pipes = sessions_with_completed_pipes["ses-01"]
```

Comment:
let's move that above the assert section. Best to keep setup, act, assert separately

Reply:
The issue with moving it is that one of things we are asserting is that "ses-01" is the only session detected as having completed pipelines. So, if `sessions_with_completed_pipes` does _not_ include `ses-01`, we ideally want this test to fail at
```python
assert sessions_with_completed_pipes.keys() == {"ses-01"}
``` 
and not
```python
ses01_completed_pipes = sessions_with_completed_pipes["ses-01"]
```

the second statement is really just a convenience variable to avoid having to repeat `sessions_with_completed_pipes["ses-01"]` in the remaining assertion statements. To avoid having a variable definition after the assertion, I could just remove the first assert statement and rely on the variable definition line failing when the key does not exit, but I'm not sure that's much better?

Reply:
Why not use `.get("ses-01") instead of using the key? The point here is really more about making the test easy to read my sticking with a standard structure and keeping things separated into sections. 

The order of the line is not a huge deal here, but neither is moving it up one line I think.